src/wp-includes/blocks/navigation-link.php

```diff
@@ -323,12 +323,11 @@ function build_variation_for_navigation_link( $entity, $kind ) {
 }
 
 /**
- * Register the navigation link block.
+ * Returns an array of variations for the navigation link block.
  *
- * @uses render_block_core_navigation()
- * @throws WP_Error An WP_Error exception parsing the block definition.
+ * @return array
  */
-function register_block_core_navigation_link() {
+function build_navigation_link_block_variations() {
```

Comment:
Setting aside the fact that these will need to be changed in the Gutenberg repo, rather than directly in core, all of these callback functions should probably be updated to include "core", like `build_core_navigation_link_block_variations`, etc., to avoid potential conflicts with any third party blocks.

Reply:
There is a linting rule in the Gutenberg plugin that should enforce using a proper prefix for the function.

Reply:
@joemcgill my only concern about changing name is consistency in naming convention. 

Giving that `build_template_part_block_variations` is an older public function, moving to `build_core_*` means we might have to depreciate `build_template_part_block_variations` with a wrapper code calling `build_core_template_part_block_variations`.

Reply:
Ah, that makes sense. I hadn't thought about this already being in core with this name. It's somewhat unfortunate naming, but probably fine. We can raise this concern in the related Gutenberg PR where these changes will need to be made. Not a blocker for this effort.